doc/nrf/gs_testing.rst

```diff
@@ -120,3 +120,105 @@ To connect to the nRF9160-based kit with LTE Link Monitor, perform the following
+
+Debugging secure and non-secure domains
+=======================================
+
+When using an ``_ns`` build target, by default you can only debug firmware in the non-secure part of the application core firmware.
+
+To debug firmware running in the secure domain, you need to build Trusted Firmware-M with debug symbols enabled and load the symbols during the debugging session.
+To build Trusted Firmware-M with debug symbols, set the :kconfig:option:`CONFIG_TFM_CMAKE_BUILD_TYPE_RELWITHDEBINFO` Kconfig option.
+To load the Trusted Firmware-M debug symbols in the GDB tool, run the following command:
+
+.. code-block:: console
+
+   (gdb) add-symbol-file build/tfm/bin/tfm_s.elf
+
+For more information about the GDB tool, see `Debug tools`_.
+
+Debug configuration
+===================
+
+The following Kconfig options are automatically set by the |VSC| debugger in the |nRFVSC| when you enable the debug configuration before building the application:
+
+* :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` set to ``y``
+* :kconfig:option:`CONFIG_DEBUG_THREAD_INFO` set to ``y``
+
+There are many more Kconfig options for debugging that are specific to different modules.
+For details, see the respective documentation pages of the modules.
+
+Debug build types
+-----------------
+
+Some applications and samples provide a specific build type that enables additional debug functionalities.
+You can select build types when you are :ref:`configuring the build settings <gs_modifying_build_types>`.
+
+Debug logging
+-------------
+
+You can use logging system to get more information about the state of your application.
+Logs are integrated to many various modules and subsystems in the |NCS| and Zephyr.
+These logs are visible once you configure logger for your application.
+
+You can also configure log level per logger module, for example to get more information about a given subsystem.
+See :ref:`ug_logging` for details on how to enable and configure logs.
+
+Debug libraries
+---------------
+
+The |NCS| also provides several libraries and drivers for debugging different components.
+For example:
+
+* You can use :ref:`lib_debug` for any of your applications, for example to measure CPU load or trace hardware peripheral events on pins.
+
+* You can use :ref:`nrf_profiler` to measure performance and debug applications without introducing big performance overhead.
+  This option requires introducing additional code changes: your application must register profiler events and log their occurrences.
+
+* When working with the :ref:`SEGGER J-Link with the RTT feature <testing_rtt_connect>`, you can use the :ref:`lib_eth_rtt`, which is useful for handling data transfer.
+
+* The Thread protocol implementation offers :ref:`pre-built libraries with debug symbols <thread_ug_feature_updating_libs>`.
+
+* The Zigbee protocol implementation offers :ref:`lib_zigbee_osif` and :ref:`lib_zigbee_shell` with custom Kconfig options that you can set for debugging.
+
```

Comment:
There is also Memfault SDK maybe it is worth mentioning here, https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/libraries/others/memfault_ncs.html#mod-memfault.
What do you think ?

Reply:
Discussed adding this with @trond-snekvik and we opted to skip this mention, because it is not exactly related to debugging. It can help, but we didn't think it is something we should highlight among more debug-relevant items.